shared/router-v2/header/syncing-folders.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import {namedConnect} from '../../util/typed-connect'
+import PieSlice from '../../fs/common/pie-slice'
+
+type Props = {|
+  progress: number,
+|}
+
+const SyncingFolders = (props: Props) =>
+  props.progress !== 1.0 && (
+    <Kb.Box2 direction="horizontal" alignItems="center">
+      <PieSlice degrees={props.progress * 360} animated={true} />
+      <Kb.Text type="BodyTiny" style={{marginLeft: 5}}>
+        Syncing Folders...
+      </Kb.Text>
+    </Kb.Box2>
+  )
+
+const mapStateToProps = state => {
+  if (state.fs.syncingFoldersProgress.bytesTotal === 0) {
+    return {progress: 1}
+  }
+  return {progress: state.fs.syncingFoldersProgress.bytesFetched / state.fs.syncingFoldersProgress.bytesTotal}
```

Comment:
It probably doesn't make a huge difference in this particular case, but it'd be nice to just return `state.fs.syncingFoldersProgress` here and do this work in `mergeProps`. `mapStateToProps` needs to be fast as all connected components need to be iterated through on state updates.